pool.go

```diff
@@ -0,0 +1,192 @@
+package server
+
+import (
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/bblfsh/sdk/protocol"
+	"github.com/pkg/errors"
+	"golang.org/x/time/rate"
+)
+
+const (
+	DefaultPoolMin               = 1
+	DefaultPoolMax               = 5
+	DefaultPoolTimeBeforeNew     = time.Second * 3
+	DefaultPoolTimeBetweenSpawns = time.Second * 10
+	DefaultPoolTimeBeforeClean   = time.Minute * 1
+)
+
+// DriverPool controls a pool of drivers and balances requests among them,
+// ensuring each driver does not get concurrent requests.
+type DriverPool struct {
+	// Min
+	Min               int
+	Max               int
+	TimeBeforeNew     time.Duration
+	TimeBetweenSpawns time.Duration
+	TimeBeforeClean   time.Duration
+	New               func() (Driver, error)
+
+	cur     int
+	ch      chan Driver
```

Comment:
call it `ready` and I think it should probably be a buffered driver, with enough space to hold MAX drivers.